pkg/daemon/daemon.go

```diff
@@ -1971,6 +2111,30 @@ func (dn *Daemon) completeUpdate(desiredConfigName string) error {
 	return nil
 }
 
+func (dn *Daemon) triggerUpdate(currentConfig, desiredConfig *mcfgv1.MachineConfig, currentImage, desiredImage string) error {
+	// If both of the image annotations are empty, this is a regular MachineConfig update.
+	if desiredImage == "" && currentImage == "" {
+		return dn.triggerUpdateWithMachineConfig(currentConfig, desiredConfig, true)
+	}
+
+	// If the desired image annotation is empty, but the current image is not
+	// empty, this should be a regular MachineConfig update.
+	//
+	// However, the node will not roll back from a layered config to a
+	// non-layered config without admin intervention; so we should emit an error
+	// for now.
+	if desiredImage == "" && currentImage != "" {
```

Comment:
@yuqi-zhang Without adding this check, `dn.updateImage()` would attempt to use an empty string (`""`) as an image pullspec, which fails for obvious reasons. I'm not sure if this is a reasonable thing to do here until we can get rollbacks to work correctly.

Reply:
I think this is fair as a tech preview and telling the users to create a custom pool for this. I think we just have to be explicit in telling the users that marking a node for layering is non-reversible in our docs.

We could potentially retroactively fix this as a "bug" as well so I am not too worried about this